### PR TITLE
Docs Restructuring, Edits, and Additions

### PR DESCRIPTION
**Pull request by hasanza**
This PR restructures and edits the docs to make them more accurate, detailed, and approachable for users. This change is part of a wider goal of making the docs an exhaustive collection of highly readable resources the users can use to learn how to do stuff and troubleshoot their issues. This is only the first batch of changes with more changes incoming. Later PRs are likely to be smaller compared to this one.

While a lot of the files have been edited, changes are mostly concentrated in the "requirements" page and the "getting started" page. Work continues on the rest.

### CODE REVIEW

File: docs/start/considerations.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+### PoS Data
+
+You only need to setup proof of space data once to smesh for an unlimited period of time. The larger your proof of space data is, the higher your smeshing rewards will be.
+
+You can use any magnetic hard disk drive (HDD) or solid state drive (SSD) for storing the PoS data. There is no significant advantage in using an SSD over HDD. 
+
+You do not need to have temporary disk space larger than your final proof of space data. Creating proof of space data on more than one volume is not yet supported.
+
+### Network Operations
+
```
Comment (pigmej) [inline docs/start/considerations.md:31]: from where are these numbers?

File: docs/start/considerations.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+The biggest resource consumed by a Spacemesh node is bandwidth. A full node currently consumes between 150-1500mb/hour of bandwidth. The node is quiescent most of the time, consuming only 100-300kbit/s, but spikes as high as 35Mbit/s every few minutes when a new layer appears and network traffic increases (Hare messages, proposals, etc.). Bandwidth is variable and depends heavily on the network condition and the number and quality of one's peers.
```
Comment (pigmej) [inline docs/start/considerations.md:32]: I think given that compact encoding is on the way and hare committee reduction, too, we can skip that here.

File: docs/start/considerations.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+As a benchmark, the minimum PoST data size (256GiB, or 4 storage units) can be initialized in 8-10 hours using a single Nvidia GeForce RTX 4090. The same data would take around 36 hours using a Nvidia GeForce RTX 3060. Initialization time scales linearly, so that initializing 40 SU would take 10x as long as initializing 4 SU.
+
+Note that initialization can be parallelized across multiple systems or multiple GPUs, and it can be performed on one system (including in the cloud) and the resultant data moved to another system (e.g., one without a GPU) for long-term storage and mining. There's more information on this below.
+
+### Ongoing Mining
+
+Mining on an ongoing basis does _not_ require a GPU. The only additional resource consideration for mining, beyond the baseline cost of running a full node (as outlined above), is the cost of generating a [proof of space](./../learn/post.md) once per epoch. The time required to generate a proof is complicated and multivariate, depending on several factors including CPU speed, disk speed, size of PoST data, and the `smeshing-proving-opts` settings specified in config (more on this below).
```
Comment (pigmej) [inline docs/start/considerations.md:48]: I believe we always refer to Mining as Smeshing. Would probably make sense to stick to it.

File: docs/start/considerations.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+The general requirement is a modern CPU with the [AES-NI](https://en.wikipedia.org/wiki/AES_instruction_set) instruction and a disk that's fast enough to perform a single, sequential read over the entire PoST data in 4-5 hours. This allows enough time to perform a second read during the PoET cycle gap on the off chance that the first pass fails to generate a valid proof. Hard drive performance varies widely but most modern 7200 RPM HDDs can perform a sequential read at 100-200mb/sec (5400 RPM drives are slower). This means that, to be safe, a smesher shouldn't initialize more than 2-4 TB or 8-16 SU on a single HDD. It's certainly possible to initialize more and smeshers with 4TB drives and fast CPUs report no issues generating a proof in time, but the probability of failure increases with each additional byte initialized. It's also possible to achieve much higher read speeds using more expensive technology such as SSD and/or RAID.
```
Comment (pigmej) [inline docs/start/considerations.md:50]: That 4-5 hours is very misleading. One can set up higher nonces etc. Probably makes sense to rework that part.

File: docs/start/smeshing/smeshing_adv/setup.md [hunk 5]
```diff
@@ -2,8 +2,29 @@
...
+
+The one piece of dynamic data that you need to perform PoST init is the highest commitmentAtxId. This is simply the ID of the highest ATX known at the time that PoST initialization begins. It is used to mitigate a certain class of protocol attack; for more information, see POPS-VRF implementation.
+The simplest way to get this is to ask your own node, using the API as such:
+
+```bash
+> grpcurl -plaintext 127.0.0.1:9092 spacemesh.v1.ActivationService.Highest | jq -r '.atx.id.id' | base64 -d | xxd -p -c 64
+435fa442517e9c75087de1b06d2a9d12c345505f3cac93ac52b816171ce48308
+```
+
+While we plan to operate public nodes in the near future, we do not operate any at present. Therefore, we strongly encourage you to double-check this information, ideally by running your own node, or alternatively by checking with the community on Discord.
```
Comment (pigmej) [inline docs/start/smeshing/smeshing_adv/setup.md:23]: mainnet-api is available so we could put it here too. Just I think v1 is not supported there. @kacpersaw is there equivalent of `spacemesh.v1.ActivationService.Highest` in v2?

File: docs/start/smeshing/start.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+Smapp also lets you customize your config file and use the node API directly. Running Smapp does not provide any greater or lesser functionality than running go-spacemesh directly, but some technical users who prefer working with a CLI rather than a GUI may wish to run `go-spacemesh` as a system service, etc. Users running without a GUI will of course have no choice but to run `go-spacemesh` directly via CLI.
+
+### CLI - Advanced Smeshing
+
+You can also run a full or smeshing node entirely from the command line using [go-spacemesh](https://github.com/spacemeshos/go-spacemesh). In addition, the [`postcli`](https://github.com/spacemeshos/post/blob/develop/cmd/postcli/README.md) tool can be used to generate PoS data with much more fine-grain control.
+
+Some advantages of smeshing using `go-spacemesh` and `postcli` over Smapp include:
+
+- `go-spacemesh`:
+    - More lightweight compared to Smapp.
```
Comment (pigmej) [inline docs/start/smeshing/start.md:68]: I think it's false :) Smapp uses go-sm behind the scenes.

File: docs/start/smeshing/start.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+    - Can run on headless (non-GUI) systems.
+    - Advanced configuration is possible. Smapp has pre-programmed values and changing them is not as straightforward as it is on `go-spacemesh`.
```
Comment (pigmej) [inline docs/start/smeshing/start.md:70]: Smapp supports config overrides, I think it's even easier in Smapp as you just put the modified values.

**Review by pigmej — CHANGES_REQUESTED** [review]
Fairly good overall, few changes needed though.

File: docs/start/considerations.md (hunk 1 above)
Comment (hasanza) [reply docs/start/considerations.md:31]: All the numbers were already there before my edits.

File: docs/start/considerations.md (hunk 3 above)
Comment (hasanza) [reply docs/start/considerations.md]: Changed all instances of "mining" to "smeshing".

File: docs/start/considerations.md (hunk 4 above)
Comment (hasanza) [reply docs/start/considerations.md]: Changed to: "The general requirement is a modern CPU with the [AES-NI](https://en.wikipedia.org/wiki/AES_instruction_set) instruction and a disk that is fast enough to perform a single, sequential read over the entire PoST data **in a reasonable amount of time**.".

File: docs/start/smeshing/start.md (hunk 6 above)
Comment (hasanza) [reply docs/start/smeshing/start.md:68]: Isn't running an electronJS + go-sm app relatively more resource-consuming vs. just running go-sm?

File: docs/start/smeshing/start.md (hunk 7 above)
Comment (hasanza) [reply docs/start/smeshing/start.md]: Removed.

File: docs/start/smeshing/smeshing_adv/setup.md (hunk 5 above)
Comment (kacpersaw) [reply docs/start/smeshing/smeshing_adv/setup.md:23]: Not yet. We can add this in the next iteration.

Comment (pigmej) [reply docs/start/smeshing/smeshing_adv/setup.md:23]: Yeah let's plan it then this documentation entry would be easier ;)

**Comment by pigmej** [comment]
My only concern is the new structure / URLs. That's up to @yaelmhoffman

**Review by pigmej — APPROVED** [review]

**Pull request merged**